### PR TITLE
[Laravel] Add RequestStaticValidateToInjectRector

### PR DESCRIPTION
**Pull request by TomasVotruba**
Ref #1138 ,
ref #290

```diff
  use Illuminate\Http\Request;
	  
  class SomeClass
  {
-     public function store()
+     public function store(\Illuminate\Http\Request $request)
      {
-         $validatedData = Request::validate(['some_attribute' => 'required']);
+         $validatedData = $request->validate(['some_attribute' => 'required']);
      }
  }
```

### CODE REVIEW

**Comment by TomasVotruba** [comment]
@mrpixeldream Could I ask you for a review? I hope I got that right :sweat_smile:

**Comment by barryvdh** [comment]
Why just the validate? Could also be for the other Requests calls, right?

**Comment by TomasVotruba** [comment]
Sure. Again, I don't use Laravel so I have no idea about the logic here :).

If you provide list of these magic static methods, that can be changed here, we can update it

**Comment by barryvdh** [comment]
Ah my bad, I guess usually it's the Request facade, not the Request class itself that is called static.
I usually prefer to inject the Request class like this ($request in the controller method), but that's already done by the other call (all facades to injection in __construct). But I would say that the Request object belongs to the current controller action usually, so it would belong there. But could be a matter of preference.

**Comment by TomasVotruba** [comment]
I'm not sure I understand. It's in controller action.

See description of this PR:

image

**Comment by barryvdh** [comment]
This one is, but not the Facade one (that is done in https://github.com/rectorphp/rector/pull/1140)

**Comment by TomasVotruba** [comment]
Why do you talk about differnt PR here?

**Comment by TomasVotruba** [comment]
Mergin for `validate` method as it is now

**Comment by mrpixeldream** [comment]
Yeah, seems fine 👍   
If you want, you can check out https://github.com/laravel/framework/blob/5.8/src/Illuminate/Support/Facades/Request.php for a list of methods that are usable on the facade and hence also on the request object. The docblock shows all of them. `Request::get()` is frequently used as well for example. I am of course open to reviewing if you want that, just wait a little longer :P

**Comment by barryvdh** [comment]
Because people usually call the Request facade, not the actual class. eg. `Request::get()` is usually an alias for `Illuminate\Support\Facades\Request`, not for `\Illuminate\Http\Request`. So those calls are already handled by the other PR (which handles all facades)

**Comment by TomasVotruba** [comment]
The code-review would be best :+1;

File: packages/Laravel/tests/Rector/StaticCall/RequestStaticValidateToInjectRector/Fixture/fixture.php.inc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+-----
+<?php
+
+namespace Rector\Laravel\Tests\Rector\StaticCall\RequestStaticValidateToInjectRector\Fixture;
+
+use Illuminate\Http\Request;
+
+class SomeClass
+{
+    public function store(\Illuminate\Http\Request $request)
```
Comment (mrpixeldream) [inline packages/Laravel/tests/Rector/StaticCall/RequestStaticValidateToInjectRector/Fixture/fixture.php.inc]: You are `use`-ing the class above already, no need to fully specify it here.

Comment (TomasVotruba) [reply packages/Laravel/tests/Rector/StaticCall/RequestStaticValidateToInjectRector/Fixture/fixture.php.inc]: This is job of coding standard. Rector is doing the hard work, ECS makes it nice again :)

See https://github.com/rectorphp/rector#how-to-apply-coding-standards

Comment (mrpixeldream) [reply packages/Laravel/tests/Rector/StaticCall/RequestStaticValidateToInjectRector/Fixture/fixture.php.inc]: Ah yeah, alright. Everything's fine then I guess 👍

**Pull request merged**